RadixWallet/Features/SettingsFeature/Preferences/Preferences+View.swift

```diff
@@ -0,0 +1,182 @@
+		}
+		#else
+		init(isDeveloperModeEnabled: Bool) {
+			self.isDeveloperModeEnabled = isDeveloperModeEnabled
+		}
+		#endif
+	}
+
+	@MainActor
+	struct View: SwiftUI.View {
+		private let store: StoreOf<Preferences>
+
+		public init(store: StoreOf<Preferences>) {
+			self.store = store
+		}
+
+		public var body: some SwiftUI.View {
+			content
+				.setUpNavigationBar(title: S.title)
+				.tint(.app.gray1)
+				.foregroundColor(.app.gray1)
+				.presentsLoadingViewOverlay()
+				.destinations(with: store)
+		}
+	}
+}
+
+extension Preferences.View {
+	@MainActor
+	private var content: some View {
+		WithViewStore(store, observe: \.viewState, send: { .view($0) }) { viewStore in
+			ScrollView {
+				VStack(spacing: .zero) {
+					ForEach(rows(viewStore: viewStore)) { kind in
+						kind.build(viewStore: viewStore)
+					}
+				}
+			}
+			.background(Color.app.gray4)
+			.onAppear {
+				viewStore.send(.appeared)
+			}
+			#if DEBUG
+			.sheet(item: viewStore.binding(get: \.exportLogsUrl, send: { _ in .exportLogsDismissed })) { item in
+					ShareView(items: [item])
+				}
+			#endif
+		}
+	}
+
+	@MainActor
+	private func rows(viewStore: ViewStoreOf<Preferences>) -> [SettingsRow<Preferences>] {
+		var visibleRows = normalRows(viewStore: viewStore)
+		#if DEBUG
+		visibleRows.append(.model(
+			title: "Export logs",
+			subtitle: "Export and save debugging logs",
+			icon: .asset(AssetResource.appSettings),
+			action: .exportLogsButtonTapped
+		))
+		#endif
+		return visibleRows
+	}
+
+	@MainActor
+	private func normalRows(viewStore: ViewStoreOf<Preferences>) -> [SettingsRow<Preferences>] {
+		[
+			.separator,
+			.model(
+				title: S.DepositGuarantees.title,
+				subtitle: S.DepositGuarantees.subtitle,
+				icon: .asset(AssetResource.depositGuarantees),
+				action: .depositGuaranteesButtonTapped
+			),
+			.model(
+				title: S.HiddenEntities.title,
+				subtitle: S.HiddenEntities.subtitle,
+				icon: .systemImage("eye.fill"),
+				action: .hiddenEntitiesButtonTapped
+			),
+			.header(S.advancedPreferences),
+			.model(
+				title: S.gateways,
+				icon: .asset(AssetResource.gateway),
+				action: .gatewaysButtonTapped
+			),
+			.custom(developerMode(viewStore: viewStore), id: "developerMode"),
+		]
+	}
+
+	private func developerMode(viewStore: ViewStoreOf<Preferences>) -> AnyView {
+		ToggleView(
+			icon: AssetResource.developerMode,
+			title: S.DeveloperMode.title,
+			subtitle: S.DeveloperMode.subtitle,
+			minHeight: .zero,
+			isOn: viewStore.binding(
+				get: \.isDeveloperModeEnabled,
+				send: { .developerModeToogled($0) }
+			)
+		)
+		.padding(.horizontal, .medium3)
+		.padding(.vertical, .medium1)
+		.background(Color.app.white)
+		.withSeparator
+		.eraseToAnyView()
+	}
+}
+
+private extension StoreOf<Preferences> {
+	var destination: PresentationStoreOf<Preferences.Destination> {
+		func scopeState(state: State) -> PresentationState<Preferences.Destination.State> {
+			state.$destination
+		}
```

Comment:
Just FYI: The only reason we use this pattern is because XCode get incredibly slow when it has to typecheck it otherwise, to the point of being unusable.

Reply:
Yeah, I learned that from having my own personal Xcode freeze experience 😁  thanks